src/test/java/org/broadinstitute/hellbender/utils/IndexUtilsUnitTest.java

```diff
@@ -0,0 +1,100 @@
+
+    @DataProvider(name= "okFeatureFiles")
+    public Object[][] okFeatureFiles() {
+        return new Object[][] {
+                { new File(getToolTestDataDir(), "test_variants_for_index.vcf")},
+                { new File(getToolTestDataDir(), "test_variants_for_index.gvcf.vcf")},
+                { new File(getToolTestDataDir(), "test_bed_for_index.bed")},
+        };
+    }
+
+    @Test(dataProvider = "okFeatureFiles")
+    public void testLoadIndex(final File featureFile) throws Exception {
+        final Index index = IndexUtils.loadTribbleIndex(featureFile);
+        Assert.assertNotNull(index);
+    }
+
+    @DataProvider(name= "okFeatureFilesTabix")
+    public Object[][] okFeatureFilesTabix() {
+        return new Object[][] {
+                { new File(getToolTestDataDir(), "test_variants_for_index.vcf.bgz")},
+        };
+    }
+
+    @Test(dataProvider = "okFeatureFilesTabix")
+    public void testLoadTabixIndex(final File featureFile) throws Exception {
+        final Index index = IndexUtils.loadTabixIndex(featureFile);
+        Assert.assertNotNull(index);
+    }
+
+    @DataProvider(name= "failTabixIndexFiles")
+    public Object[][] failTabixIndexFiles() {
+        return new Object[][] {
+                { new File(getToolTestDataDir(), "test_variants_for_index.vcf")},
+                { new File(getToolTestDataDir(), "test_variants_for_index.gvcf.vcf")},
+                { new File(getToolTestDataDir(), "test_bed_for_index.bed")},
+        };
+    }
+
+    @Test(dataProvider = "failTabixIndexFiles")
+    public void testFailLoadTabixIndex(final File featureFile) throws Exception {
+        final Index index = IndexUtils.loadTabixIndex(featureFile);
+        Assert.assertNull(index);
+    }
+
+    @DataProvider(name= "failTribbleIndexFiles")
+    public Object[][] failTribbleIndexFiles() {
+        return new Object[][] {
+                { new File(getToolTestDataDir(), "test_variants_for_index.vcf.bgz")},
+        };
+    }
+
+    @Test(dataProvider = "failTribbleIndexFiles")
+    public void testFailLoadTribbleIndex(final File featureFile) throws Exception {
+        final Index index = IndexUtils.loadTribbleIndex(featureFile);
+        Assert.assertNull(index);
+    }
+
+    @Test
+    public void testLoadIndexAcceptOldIndex() throws Exception {
+        final File featureFile = new File(getToolTestDataDir(), "test_variants_for_index.newerThanIndex.vcf");
+        final File featureFileIdx = new File(getToolTestDataDir(), "test_variants_for_index.newerThanIndex.vcf.idx");
+        final File tmpDir = BaseTest.createTempDir("testLoadIndexAcceptOldIndex");
+
+        Files.copy(featureFile.toPath(), tmpDir.toPath().resolve(featureFile.toPath().getFileName()));
+        Files.copy(featureFileIdx.toPath(), tmpDir.toPath().resolve(featureFileIdx.toPath().getFileName()));
+        featureFile.setLastModified(System.currentTimeMillis()); //touch the file but the index
```

Comment:
So, you went to all this trouble to make a copy of `featureFile` in a temp dir, and then you go ahead and modify the original anyway! :) Change this to modify the copy in `tmpDir`, not the checked-in file.